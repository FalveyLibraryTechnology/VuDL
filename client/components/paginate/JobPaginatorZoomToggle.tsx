import React from "react";

import { usePaginatorContext } from "../../context/PaginatorContext";
import PaginatorPreview from "./PaginatorPreview";
import PaginatorZoomy from "./PaginatorZoomy";

const JobPaginatorZoomToggle = () => {
    const {
        state: { order, zoom, currentPage },
        action: { getJobImageUrl },
    } = usePaginatorContext();

    if (order.length < 1) {
        return <div>Preview not available.</div>;
    }
    return zoom ? (
        <PaginatorZoomy img={getJobImageUrl(order[currentPage], "large")} />
    ) : (
        <PaginatorPreview img={getJobImageUrl(order[currentPage], "medium")} />
    );
};

export default JobPaginatorZoomToggle;
