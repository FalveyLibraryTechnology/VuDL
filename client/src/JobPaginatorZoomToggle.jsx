import React from "react";
import PropTypes from "prop-types";

import PaginatorPreview from "./PaginatorPreview";
import PaginatorZoomy from "./PaginatorZoomy";

const JobPaginatorZoomToggle = ({ zoom, getJobImageUrl }) => {
    return zoom ? (
        <PaginatorZoomy img={getJobImageUrl("large")} />
    ) : (
        <PaginatorPreview img={getJobImageUrl("medium")} />
    );
};

JobPaginatorZoomToggle.propTypes = {
    zoom: PropTypes.bool,
    getJobImageUrl: PropTypes.func,
};

export default JobPaginatorZoomToggle;
