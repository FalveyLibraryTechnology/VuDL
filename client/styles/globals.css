:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
