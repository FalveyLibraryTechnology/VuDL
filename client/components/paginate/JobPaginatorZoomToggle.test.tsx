import React from "react";
import { beforeEach, describe, expect, it, jest } from "@jest/globals";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import JobPaginatorZoomToggle from "./JobPaginatorZoomToggle";

const mockUseJobPaginatorContext = jest.fn();
jest.mock("../../context/PaginatorContext", () => ({
    usePaginatorContext: () => {
        return mockUseJobPaginatorContext();
    },
}));
jest.mock("./PaginatorZoomy", () => () => "PaginatorZoomy");
jest.mock("./PaginatorPreview", () => () => "PaginatorPreview");

describe("JobPaginatorZoomToggle", () => {
    let paginatorValues;
    beforeEach(() => {
        paginatorValues = {
            state: {
                order: [
                    {
                        filename: "test.jpg",
                    },
                ],
                zoom: true,
                currentPage: 0,
            },
            action: {
                getJobImageUrl: jest.fn(),
            },
        };
        mockUseJobPaginatorContext.mockReturnValue(paginatorValues);
    });

    it("renders", () => {
        const wrapper = shallow(<JobPaginatorZoomToggle />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders preview not available", () => {
        paginatorValues.state.order = [];
        const wrapper = shallow(<JobPaginatorZoomToggle />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.text().includes("Preview not available")).toBeTruthy();
    });

    it("renders PaginatorZoomy", () => {
        const wrapper = mount(<JobPaginatorZoomToggle />);

        expect(wrapper.contains("PaginatorZoomy")).toBeTruthy();
        expect(paginatorValues.action.getJobImageUrl).toHaveBeenCalledWith(paginatorValues.state.order[0], "large");
    });

    it("renders PaginatorPreview", () => {
        paginatorValues.state.zoom = false;
        const wrapper = mount(<JobPaginatorZoomToggle />);

        expect(wrapper.contains("PaginatorPreview")).toBeTruthy();
        expect(paginatorValues.action.getJobImageUrl).toHaveBeenCalledWith(paginatorValues.state.order[0], "medium");
    });
});
