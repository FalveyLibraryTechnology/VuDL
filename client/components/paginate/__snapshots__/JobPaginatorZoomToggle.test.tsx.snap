// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPaginatorZoomToggle renders 1`] = `<Component />`;

exports[`JobPaginatorZoomToggle renders preview not available 1`] = `
<div>
  Preview not available.
</div>
`;
